frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/upgrade/KafkaConfigUpgrade.java

```diff
@@ -0,0 +1,369 @@
+
+                LOGGER.info("old disk resource : {}", resource);
+
+                //volume.toBuilder() was complaining that mode is not set, so setting everything manually
+                Protos.Resource.DiskInfo diskInfo = resource.getDisk();
+                Protos.Volume volume = diskInfo.getVolume();
+                Protos.Volume.Builder volumeBuilder = volume.toBuilder();
+                Protos.Volume newVolume = volumeBuilder.setSource(volume.getSource())
+                        .setContainerPath(this.newPath) //kafka-broker-data
+                        .setMode(volume.getMode())
+                        .build();
+
+                Protos.Resource.DiskInfo.Builder newDiskInfoBuilder = diskInfo.toBuilder();
+                Protos.Resource.DiskInfo newDiskInfo = newDiskInfoBuilder.setVolume(newVolume)
+                        .setPersistence(diskInfo.getPersistence())
+                        .build();
+
+                Protos.Resource.Builder resourceBuilder = resource.toBuilder();
+                Protos.Resource newResource = resourceBuilder.setDisk(newDiskInfo)
+                        .setName(resource.getName())
+                        .setReservation(resource.getReservation())
+                        .setRole(resource.getRole())
+                        .setScalar(resource.getScalar())
+                        .build();
+
+                LOGGER.info("new disk resource : {}", newResource);
+
+                resourcesList.add(newResource);
+            }
+            taskInfoBuilder.clearResources();
+            taskInfoBuilder.addAllResources(resourcesList);
+
+            Protos.TaskInfo newTaskInfo = taskInfoBuilder.build();
+            LOGGER.info("Task {} new TaskInfo: ", newTaskInfo.getName(), newTaskInfo);
+            taskInfoList.add(newTaskInfo);
+        }
+        return taskInfoList;
+    }
+
+    private Map<String, Protos.TaskStatus> generateStatusMap(Collection<String> oldTaskNames)
+            throws KafkaConfigUpgradeException{
+        Map<String, Protos.TaskStatus> taskStatusMap = new HashMap<>();
+
+        for (String oldTaskName : oldTaskNames) {
+            int brokerId = oldTaskName2BrokerId(oldTaskName);
+            String newName = getNewTaskName(brokerId);
+            Optional<Protos.TaskStatus> optionalStatus = stateStore.fetchStatus(oldTaskName);
+            if (!optionalStatus.isPresent()){
+                throw new KafkaConfigUpgradeException("Can not fetch status for Task " + oldTaskName);
+            }
+            taskStatusMap.put(newName, optionalStatus.get());
+        }
+        return taskStatusMap;
+    }
+
+    private int oldTaskName2BrokerId(String taskName){
+        Pattern pattern = Pattern.compile("(.*)-(\\d+)");
+        Matcher matcher = pattern.matcher(taskName);
+        matcher.find();
+        return Integer.parseInt(matcher.group(2));
+    }
+
+    private String getNewTaskName(int brokerID){
+        return "kafka-" + brokerID + "-broker"; //kafka-2-broker
```

Comment:
This is incorrect. `kafka` cannot be hard-coded. It needs to be the name of service, afaik.